### DISCUSSION TITLE
Question about installing self-built executable on Kobo

### DISCUSSION THREAD

**Discussion by rexkerr** [koreader/koreader#7374] [Q&A] [answered]
Out of an abundance of caution, I want to ask before blindly jumping in and potentially bricking my reader.

I want to continue to improve the warm light ramp-down that didn't work out as expected in my recent PR, but would like to test locally before pushing anything to avoid breaking the functionality for others.

I've switched over to a Linux configuration, and built the toolchain repo and this one.  I then ran `kodev release kobo` and I get a zip file something like koreader-kobo-v2021.02-39-g5303165b_2021-03-01.zip, so I think that the setup is working.   Is that zip file the one that I want to put on the device for it to auto-install, or do I need to do something more to get an installable package?

**Comment by NiLuJe** [comment]
If by auto-install you mean the OCP, then nope. But you can just unzip it to `/mnt/onboard/.adds` ;).

**Comment by NiLuJe** [comment, marked as answer]
Note that unless you're actually modifying C or C++ modules, you don't need a custom build to test stuff.

If you're only tweaking Lua code, just pushing that works just fine.

e.g., I routinely do something like

`for file in $(git diff --name-only upstream/master) ; do ; [[ -f "${file}" ]] && scp "${file}" "root@kindle:/mnt/onboard/.adds/koreader/${file}" ; done`

(Yeah, `/etc/hosts` says that an USBNet device is called kindle, don't mind me ;p).

**Reply by rexkerr** [reply]
Thanks.  I'll look into that -- makes sense that the interpreted code doesn't need a full build.  Since my changes are limited to an individual file, this should be very easy to test!

**Reply by NiLuJe** [reply]
Obviously, just restart KOReader after pushing the files ;).

**Reply by Frenzie** [reply]
To catch a few more changed files:
```
for file in $({ git diff --name-only; git diff --name-only origin/master; } | sort | uniq); do [[ -f "${file}" ]] && scp "${file}" "asdf@kj.kj.be:/data/global/kj/${file}"; done && ssh asdf@kj.kj.be
```
